optuna_dashboard/ts/components/PreferentialTrials.tsx

```diff
@@ -1,37 +1,263 @@
+    const outputComponent: FeedbackComponentType =
+      outputComponentType === "note"
+        ? ({ output_type: "note" } as FeedbackComponentNote)
+        : ({
+            output_type: "artifact",
+            artifact_key: artifactKey,
+          } as FeedbackComponentArtifact)
+    actions.updateFeedbackComponent(studyDetail.id, outputComponent)
+  }
+
+  return (
+    <ModalPage displayFlag={settingShown} onClose={onClose}>
+      <Typography
+        variant="h4"
+        sx={{
+          padding: theme.spacing(2),
+          fontWeight: theme.typography.fontWeightBold,
+        }}
+      >
+        Settings
+      </Typography>
+      <Box
+        sx={{
+          padding: theme.spacing(2),
+          display: "flex",
+          flexDirection: "column",
+        }}
+      >
+        <FormControl component="fieldset">
+          <FormLabel component="legend">Output Component:</FormLabel>
+          <Select
+            value={outputComponentType}
+            onChange={(e) => {
+              setOutputComponentType(e.target.value as "note" | "artifact")
+            }}
+          >
+            <MenuItem value="note">Note</MenuItem>
+            <MenuItem value="artifact">Artifact</MenuItem>
+          </Select>
+        </FormControl>
+        {outputComponentType === "artifact" ? (
+          <FormControl
+            component="fieldset"
+            disabled={studyDetail.union_user_attrs.length === 0}
+          >
+            <FormLabel component="legend">
+              User Attribute Key Corresponding to Output Artifact Id:
+            </FormLabel>
+            <Select
+              value={
+                studyDetail.union_user_attrs.length !== 0
+                  ? artifactKey
+                  : "error"
+              }
+              onChange={(e) => {
+                setArtifactKey(e.target.value)
+              }}
+            >
+              {studyDetail.union_user_attrs.length === 0 ? (
+                <MenuItem value="error">No user attributes</MenuItem>
+              ) : null}
+              {studyDetail.union_user_attrs.map((attr, index) => {
+                return (
+                  <MenuItem key={index} value={attr.key}>
+                    {attr.key}
+                  </MenuItem>
+                )
+              })}
+            </Select>
+          </FormControl>
+        ) : null}
+      </Box>
+    </ModalPage>
+  )
+}
+
+const isComparisonReady = (
+  trial: Trial,
+  componentType: FeedbackComponentType
+): boolean => {
+  if (componentType === undefined || componentType.output_type === "note") {
+    return trial.note.body !== ""
+  }
+  if (componentType.output_type === "artifact") {
+    const artifactId = trial?.user_attrs.find(
+      (a) => a.key === componentType.artifact_key
+    )?.value
+    const artifact = trial?.artifacts.find((a) => a.artifact_id === artifactId)
+    return artifact !== undefined
+  }
+  return false
+}
+
+export const OutputContent: FC<{
```

Comment:
How about splitting the file to PreferentialOutputContent since this component is imported from multiple files.